Formula/opencascade.rb

```diff
@@ -0,0 +1,67 @@
+class Opencascade < Formula
+  desc "3D modeling and numerical simulation software for CAD/CAM/CAE"
+  homepage "https://www.opencascade.com/content/core-technology"
+  url "http://git.dev.opencascade.org/gitweb/?p=occt.git;a=snapshot;h=refs/tags/V7_2_0;sf=tgz"
```

Comment:
You'll need to set the version manually.

```
Josephs-MacBook-Pro:~ joe$ brew irb
Ve==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
irb(main):001:0> Version.detect("http://git.dev.opencascade.org/gp=occt.git;a=snapshot;h=refs/tags/V7_2_0;sf=tgz", {})
=> #<Version::FromURL:0x00007fb4b148cf20 @version="2">
irb(main):002:0>
```
So
```
version "7.2.0"
```

Reply:
Will do.  Thanks for the irb tip/tutorial too.

Reply:
Question - have you guys considered adding a version validity check (e.g. semantic version pattern) to `brew audit --strict`?

Reply:
No but that is a good idea if you want to open a PR to Homebrew/brew

Reply:
Perhaps something like this would work?
```
[blacey@bbl /usr/local/Homebrew/Library/Homebrew/dev-cmd (master *)]$ git diff
diff --git a/Library/Homebrew/dev-cmd/audit.rb b/Library/Homebrew/dev-cmd/audit.rb
index 89ab79996..671133cb2 100644
--- a/Library/Homebrew/dev-cmd/audit.rb
+++ b/Library/Homebrew/dev-cmd/audit.rb
@@ -865,8 +865,8 @@ class ResourceAuditor
       problem "version #{version} should not have a leading 'v'"
     end
 
-    return unless version.to_s =~ /_\d+$/
-    problem "version #{version} should not end with an underline and a number"
+    return if version.to_s =~ /^HEAD$|^DEVEL$|^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$/
+    problem "version \"#{version}\" must conform to semantic versioning requirements"
   end
 
   def audit_download_strategy
```

Example 1 (removed the Version declaration):
```
[blacey@bbl /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula (master *)]$ brew audit --strict opencascade
opencascade:
  * Stable: version "2" must conform to semantic versioning requirements
  * stable version should not decrease (from 7.2.0 to 2)
Error: 2 problems in 1 formula
```

Example 2 (add Version "7.2.0")
```
[blacey@bbl /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula (master)]$ brew audit --strict opencascade
[blacey@bbl /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula (master)]$
```

That said, this could be a disruptive change if there are many existing formula that don't currently comply with semantic versioning requirements.